Add support for passing tools through the Configuration

Thomvis opened the pull request:
This PR adds support for passing tools through the Configuration. I built this because I wanted to run Open Deep Research from within my own Python application and give it my own tools, without the overhead of MCP.

rlancemartin commented:
@Thomvis thank you! we recently merged some updates. you can please resolve conflicts? we also may want the ability to update the research system prompt [here](https://github.com/langchain-ai/open_deep_research/blob/c84ad979d5665213ed3e22801b0a73838f314bb1/src/open_deep_research/prompts.py#L138) with configurable instructions re: the new tools.

Thomvis commented:
I've rebased my work on top of `main` and added a `custom_tools_prompt` property to `Configuration` that can be used to add info about the custom tools to the research system prompt.
